src/components/Modal/ProfileModal.tsx

```diff
@@ -8,8 +8,38 @@ export class ProfileModal extends React.Component<{
   close: Function;
   user: firebase.User;
   userImage: string | null;
+  isSubscriber?: boolean;
+  discordUsername?: string;
+  discordDiscriminator?: string;
 }> {
-  public state = { resetDisabled: false, deleteConfirmOpen: false };
+  public state = {
+    resetDisabled: false,
+    deleteConfirmOpen: false,
+  };
+
+  authDiscord = () => {
+    const url = process.env.REACT_APP_DISCORD_AUTH_URL;
+    window.open(
+      url,
+      '_blank',
+      'toolbar=0,location=0,menubar=0,width=450,height=900'
+    );
+  };
+
+  deleteDiscord = async () => {
+    const token = await this.props.user.getIdToken();
+    await window.fetch(serverPath + '/discord/auth', {
```

Comment:
seems confusing to have the same POST call do delete. Maybe make a separate DELETE verb call to the same endpoint?